src/main/java/it/gov/pagopa/merchant/service/pointofsales/PointOfSaleServiceImpl.java

```diff
@@ -28,116 +34,182 @@
+          pointOfSale.getId(), pointOfSale.getMerchantId());
+      return;
     }
 
+    UsersResource usersResource = keycloakAdminClient.realm(realm).users();
+    List<UserRepresentation> existingUsers = usersResource.searchByEmail(contactEmail, true);
 
-    /**
-     * Verifies if the merchant exists in the system.
-     *
-     * @param merchantId the ID of the merchant to check
-     * @throws MerchantNotFoundException if the merchant does not exist
-     */
-    private void verifyMerchantExists(String merchantId){
-        MerchantDetailDTO merchantDetail = merchantService.getMerchantDetail(merchantId);
-        if(merchantDetail == null){
-            throw new MerchantNotFoundException(String.format(MerchantConstants.ExceptionMessage.MERCHANT_NOT_FOUND_MESSAGE,merchantId));
-        }
+    if (existingUsers.isEmpty()) {
+      createNewUserAndSendActionsEmail(usersResource, contactEmail, pointOfSale);
+    } else {
+      log.info("[KEYCLOAK] User with email {} already exists. The new Point of Sale with ID {} will be associated with the existing user.",
+          contactEmail, pointOfSale.getId());
     }
-
-    public PointOfSale getPointOfSaleById(String pointOfSaleId){
-        return pointOfSaleRepository.findById(pointOfSaleId)
-                .orElseThrow(() -> new PointOfSaleNotFoundException(String.format(PointOfSaleConstants.MSG_NOT_FOUND,pointOfSaleId)));
+  }
+
+  private void createNewUserAndSendActionsEmail(UsersResource usersResource, String email, PointOfSale pointOfSale) {
+    UserRepresentation newUser = new UserRepresentation();
+    newUser.setEmail(email);
+    newUser.setUsername(email);
+    newUser.setEnabled(true);
+    newUser.setEmailVerified(true);
+
+    log.info("[KEYCLOAK] Attempting to create a new Keycloak user for email {} linked to Point of Sale ID {}",
+        email, pointOfSale.getId());
+
+    try (Response response = usersResource.create(newUser)) {
+      if (response.getStatus() == Response.Status.CREATED.getStatusCode()) { // Status code 201
+        String userId = CreatedResponseUtil.getCreatedId(response);
+        log.info("[KEYCLOAK] User {} created successfully with ID {}. Sending password setup email.", email, userId);
+
+        // The action "UPDATE_PASSWORD" sends an email with a link that will expire after the lifespan to reset the user password
+        usersResource.get(userId).executeActionsEmail(keycloakClientId, redirectURI, keycloakUserActionsEmailLifespan, List.of("UPDATE_PASSWORD"));
+
+      } else {
+        // Handling non-success cases with a log
+        log.error("[KEYCLOAK] Failed to create Keycloak user for email {}. Status: {}, Reason: {}. Response body: {}",
+            email, response.getStatus(), response.getStatusInfo().getReasonPhrase(), response.readEntity(String.class));
+      }
+    } catch (Exception e) {
+      log.error("[KEYCLOAK] An exception occurred while creating Keycloak user for email {}", email, e);
+      throw e;
```

Comment:
the re-throw means that if the error on creating user fails then the point of sale is rolled back and deleted from db? In this case the next error logs in the flow are coherent?